include/exadg/operators/navier_stokes_calculators.h

```diff
@@ -72,6 +109,125 @@ class ShearRateCalculator
   unsigned int quad_index;
 };
 
+template<int dim, typename Number>
+class VorticityCalculator
+{
+public:
+  static unsigned int const number_vorticity_components = (dim == 2) ? 1 : dim;
+
+  typedef dealii::LinearAlgebra::distributed::Vector<Number> VectorType;
+
+  typedef VorticityCalculator<dim, Number> This;
+
+  typedef CellIntegrator<dim, dim, Number> CellIntegratorVector;
+
+  typedef dealii::Tensor<1, dim, dealii::VectorizedArray<Number>> vector;
+
+  VorticityCalculator();
+
+  void
+  initialize(dealii::MatrixFree<dim, Number> const & matrix_free_in,
+             unsigned int const                      dof_index_in,
+             unsigned int const                      quad_index_in);
+
+  void
+  compute_vorticity(VectorType & dst, VectorType const & src) const;
+
+private:
+  void
+  cell_loop(dealii::MatrixFree<dim, Number> const &       matrix_free,
+            VectorType &                                  dst,
+            VectorType const &                            src,
+            std::pair<unsigned int, unsigned int> const & cell_range) const;
+
+  dealii::MatrixFree<dim, Number> const * matrix_free;
+  unsigned int                            dof_index;
+  unsigned int                            quad_index;
+};
+
+template<int dim, typename Number>
+class VelocityMagnitudeCalculator
+{
+private:
+  typedef VelocityMagnitudeCalculator<dim, Number> This;
+
+  typedef dealii::LinearAlgebra::distributed::Vector<Number> VectorType;
+
+  typedef dealii::VectorizedArray<Number> scalar;
+
+  typedef std::pair<unsigned int, unsigned int> Range;
+
+  typedef CellIntegrator<dim, dim, Number> IntegratorVector;
+  typedef CellIntegrator<dim, 1, Number>   IntegratorScalar;
+
+public:
+  VelocityMagnitudeCalculator();
+
+  void
+  initialize(dealii::MatrixFree<dim, Number> const & matrix_free_in,
+             unsigned int const                      dof_index_u_in,
+             unsigned int const                      dof_index_u_scalar_in,
+             unsigned int const                      quad_index_in);
+
+  void
+  compute(VectorType & dst, VectorType const & src) const;
+
+private:
+  void
+  cell_loop(dealii::MatrixFree<dim, Number> const & matrix_free,
+            VectorType &                            dst,
+            VectorType const &                      src,
+            Range const &                           cell_range) const;
+
+  dealii::MatrixFree<dim, Number> const * matrix_free;
+
+  unsigned int dof_index_u;
+  unsigned int dof_index_u_scalar;
+  unsigned int quad_index;
+};
+
+template<int dim, typename Number, bool compressible_flow>
```

Comment:
I think `compressible_flow` should be realized as an additional parameter to the initialize function (with the goal to avoid template parameters as much as possible).

Reply:
Ok, I introduced it as a member variable and passed it as a bool const into the cell_loop, hoping that the compiler realizes that it is constant and gets rid of the if() at the quadrature point level. I rather did that than duplicating the cell_loop, which would be a safer option? What do you prefer?